resources/user-agents/browsers/firefox/firefox-generic.json

```diff
@@ -168,82 +170,28 @@
           ]
         },
         {
-          "match": "Mozilla/5.0 (#PLATFORM#rv:*) Gecko* Firefox*",
+          "match": "Mozilla/5.0 (#PLATFORM#; rv:1.9.1*) Gecko* Firefox *",
           "device": "Windows Desktop",
           "platforms": [
-            "Win10_x64", "Win10_64", "Win10_32",
-            "Win10_x64_B", "Win10_64_B", "Win10_32_B"
-          ]
-        }
-      ]
-    },
-    {
-      "userAgent": "Firefox Generic for Android",
-      "platform": "Android",
-      "engine": "Gecko",
-      "device": "general Mobile Phone (Touch)",
-      "properties": {
-        "Parent": "DefaultProperties",
-        "Comment": "Firefox Generic",
-        "Browser": "Firefox",
-        "Browser_Maker": "Mozilla Foundation",
-        "Browser_Type": "Browser",
-        "Frames": "true",
-        "IFrames": "true",
-        "Tables": "true",
-        "Cookies": "true",
-        "JavaScript": "true",
-        "CssVersion": "3"
-      },
-      "children": [
-        {
-          "match": "Mozilla/5.0 (#PLATFORM#) Gecko* Firefox/*",
-          "platforms": [
-            "AndroidMobile_4_0", "AndroidMobile_4_1", "AndroidMobile_4_2", "AndroidMobile_4_3", "AndroidMobile_4_4",
-            "AndroidMobile_5_0", "AndroidMobile_5_1",
-            "AndroidMobile"
-          ]
-        },
-        {
-          "match": "Mozilla/5.0 (#PLATFORM#) Gecko* Firefox/*",
-          "device": "general Tablet",
-          "platforms": [
-            "AndroidTablet_4_0", "AndroidTablet_4_1", "AndroidTablet_4_2", "AndroidTablet_4_3", "AndroidTablet_4_4",
-            "AndroidTablet_5_0", "AndroidTablet_5_1",
-            "AndroidTablet"
-          ]
-        },
-        {
-          "match": "Mozilla/5.0 (#PLATFORM#)*Gecko*Firefox/*",
-          "platforms": [
-            "AndroidMobile"
-          ]
-        },
-        {
-          "match": "Mozilla/5.0 (#PLATFORM#)*Gecko*Firefox/ anonymized by Abelssoft*",
+            "WinXPa_32", "Vista_32", "Win7_32", "Win8_32"
+          ],
           "properties": {
-            "isAnonymized": "true"
-          },
-          "platforms": [
-            "AndroidMobile"
-          ]
+            "Version": "3.5",
+            "MajorVer": "3",
+            "MinorVer": "5"
+          }
         },
         {
-          "match": "Mozilla/5.0 (#PLATFORM#)*Gecko*Firefox/*",
-          "device": "general Tablet",
+          "match": "Mozilla/5.0 (#PLATFORM#; rv:1.9.2*) Gecko* Firefox *",
+          "device": "Windows Desktop",
           "platforms": [
-            "AndroidTablet"
-          ]
-        },
-        {
-          "match": "Mozilla/5.0 (#PLATFORM#)*Gecko*Firefox/ anonymized by Abelssoft*",
+            "WinXPa_32", "Vista_32", "Win7_32", "Win8_32"
+          ],
           "properties": {
-            "isAnonymized": "true"
-          },
-          "device": "general Tablet",
-          "platforms": [
-            "AndroidTablet"
-          ]
+            "Version": "3.6",
```

Comment:
These are those patterns that I moved from the 3.5 and 3.6 files since they had no version tokens.